volatility3/framework/plugins/yarascan.py

```diff
@@ -56,16 +64,71 @@ def __call__(
                 for offset, name, value in match.strings:
                     yield (offset + data_offset, match.rule, name, value)
 
+    @staticmethod
+    def get_rule(rule):
+        return yara.compile(
+            sources={"n": f"rule r1 {{strings: $a = {rule} condition: $a}}"}
+        )
+
+    @staticmethod
+    def from_compiled_file(filepath):
+        return yara.load(file=resources.ResourceAccessor().open(filepath, "rb"))
+
+    @staticmethod
+    def from_file(filepath):
+        return yara.compile(file=resources.ResourceAccessor().open(filepath, "rb"))
+
+
+class YaraXScanner(interfaces.layers.ScannerInterface):
+    _version = (2, 0, 0)
+
+    # yara.Rules isn't exposed, so we can't type this properly
+    def __init__(self, rules) -> None:
+        super().__init__()
+        if rules is None:
+            raise ValueError("No rules provided to YaraScanner")
+        self._rules = rules
+
+    def __call__(
+        self, data: bytes, data_offset: int
+    ) -> Iterable[Tuple[int, str, str, bytes]]:
+        results = self._rules.scan(data)
+        for match in results.matching_rules:
+            for match_string in match.patterns:
+                for instance in match_string.matches:
+                    yield (
+                        instance.offset + data_offset,
+                        f"{match.namespace}.{match.identifier}",
+                        match_string.identifier,
+                        data[instance.offset : instance.offset + instance.length],
+                    )
+
+    @staticmethod
+    def get_rule(rule):
+        return yara_x.compile(f"rule r1 {{strings: $a = {rule} condition: $a}}")
+
+    @staticmethod
+    def from_compiled_file(filepath):
+        return yara_x.Rules.deserialize_from(
+            file=resources.ResourceAccessor().open(filepath, "rb")
+        )
+
+    @staticmethod
+    def from_file(filepath):
+        return yara_x.compile(
+            resources.ResourceAccessor().open(filepath, "rb").read().decode()
+        )
+
+
+YaraScanner = YaraXScanner if USE_YARA_X else YaraPythonScanner
+
 
 class YaraScan(plugins.PluginInterface):
     """Scans kernel memory using yara rules (string or file)."""
 
     _required_framework_version = (2, 0, 0)
     _version = (1, 3, 0)
 
-    # TODO: When the major version is bumped, take the opportunity to rename the yara_rules config to yara_string
```

Comment:
That's fine, but you then haven't bumped the plugin.  It should really become `(2, 0, 0)` for a significant bump and dependent plugins would then also need their requirements bumping.  This is to ensure that we've tested that the plugins can all work with each other...